daft/runners/pyrunner.py

```diff
@@ -67,10 +75,10 @@ def has_partition(self, idx: PartID) -> bool:
         return idx in self._partitions
 
     def __len__(self) -> int:
-        return sum(len(partition) for partition in self._partitions.values())
+        return sum(len(partition.partition()) for partition in self._partitions.values())
```

Comment:
Interesting. After this PR we actually have metadata, and don't necessarily need to reach for the partition to get the length...

Would it not be possible/safe to let `MaterializedResult.__len__` delegate appropriately between the metadata and the partition to get the length of the partition?

Reply:
I guess it doesn't really matter given that this is a local MicroPartition though